aws/resource_aws_glue_trigger.go

```diff
@@ -392,9 +426,15 @@ func expandGlueConditions(l []interface{}) []*glue.Condition {
 		m := mRaw.(map[string]interface{})
 
 		condition := &glue.Condition{
-			JobName:         aws.String(m["job_name"].(string)),
 			LogicalOperator: aws.String(m["logical_operator"].(string)),
-			State:           aws.String(m["state"].(string)),
+		}
+
+		if len(m["job_name"].(string)) > 0 {
+			condition.JobName = aws.String(m["job_name"].(string))
+			condition.State = aws.String(m["state"].(string))
+		} else if len(m["crawler_name"].(string)) > 0 {
+			condition.CrawlerName = aws.String(m["crawler_name"].(string))
+			condition.CrawlState = aws.String(m["crawl_state"].(string))
```

Comment:
Similar note here about coding defensively for missing map keys and nil values, while passing through any present configuration:

```go
if v, ok := m["crawler_name"].(string); ok && v != "" {
  condition.CrawlerName = aws.String(v)
}

if v, ok := m["crawl_state"].(string); ok && v != "" {
  condition.CrawlState = aws.String(v)
}

if v, ok := m["job_name"].(string); ok && v != "" {
  condition.JobName = aws.String(v)
}

if v, ok := m["state"].(string); ok && v != "" {
  condition.State = aws.String(v)
}
```